fastly/bigquery_test.go

```diff
@@ -0,0 +1,295 @@
+		t.Errorf("bad secret_key: %q", bq.SecretKey)
+	}
+	if bq.Format != "{\n \"timestamp\":\"%{begin:%Y-%m-%dT%H:%M:%S}t\",\n  \"time_elapsed\":%{time.elapsed.usec}V,\n  \"is_tls\":%{if(req.is_ssl, \"true\", \"false\")}V,\n  \"client_ip\":\"%{req.http.Fastly-Client-IP}V\",\n  \"geo_city\":\"%{client.geo.city}V\",\n  \"geo_country_code\":\"%{client.geo.country_code}V\",\n  \"request\":\"%{req.request}V\",\n  \"host\":\"%{req.http.Fastly-Orig-Host}V\",\n  \"url\":\"%{json.escape(req.url)}V\",\n  \"request_referer\":\"%{json.escape(req.http.Referer)}V\",\n  \"request_user_agent\":\"%{json.escape(req.http.User-Agent)}V\",\n  \"request_accept_language\":\"%{json.escape(req.http.Accept-Language)}V\",\n  \"request_accept_charset\":\"%{json.escape(req.http.Accept-Charset)}V\",\n  \"cache_status\":\"%{regsub(fastly_info.state, \"^(HIT-(SYNTH)|(HITPASS|HIT|MISS|PASS|ERROR|PIPE)).*\", \"\\\\2\\\\3\") }V\"\n}" {
+		t.Errorf("bad format: %q", bq.Format)
+	}
+	if bq.ResponseCondition != "" {
+		t.Errorf("bad response_condition: %q", bq.ResponseCondition)
+	}
+	if bq.Placement != "waf_debug" {
+		t.Errorf("bad placement: %q", bq.Placement)
+	}
+	if bq.FormatVersion != 2 {
+		t.Errorf("bad format_version: %q", bq.FormatVersion)
+	}
+
+	// List
+	var bqs []*BigQuery
+	record(t, "bigqueries/list", func(c *Client) {
+		bqs, err = c.ListBigQueries(&ListBigQueriesInput{
+			Service: testServiceID,
+			Version: tv.Number,
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(bqs) < 1 {
+		t.Errorf("bad bigqueries: %v", bqs)
```

Comment:
Feels like the message should really say 'incorrect number of bigqueries return expected X, go: Y'

Reply:
You're absolutely right! However, this PR is consistent with all other resources. I am happy to improve the client but I would prefer to discuss everything that should be refactored and then open separate PRs to improve all test suites.

Reply:
Totally agree